algebra-core/src/curves/mod.rs

```diff
@@ -308,3 +310,218 @@ where
         Fq = <Self::E1 as PairingEngine>::Fr,
     >;
 }
+
+pub trait BatchGroupArithmetic
+where
+    Self: Sized + Clone + Copy + Zero + Neg<Output = Self>,
+{
+    // This function consumes the scalars
+    // We can make this more generic in the future to use other than u16.
+
+    // TODO: Generalise to A != 0
+    // Computes [-p, p, -3p, 3p, ..., -2^wp, 2^wp]
+    fn batch_wnaf_tables(bases: &[Self], w: usize) -> Vec<Vec<Self>> {
```

Comment:
Is `w` the window size? If so, it might make it slightly clearer to rename it to `window_size` or something similar (also elsewhere in the code).


Btw, is there a link to a description of the algorithm, something that isn't just Aztec's code?